### ISSUE TITLE
converter.inference_input_type = tf.int8 is been ignored

### ISSUE THREAD

**Issue by FuchsPhi** [tensorflow/tensorflow#41697]
**System information**
- Docker image tensorflow/tensorflow:2.2.0 
- Same issue with Windows python 3 and tensorflow 2.2.0 installed via pip

**Command used to run the converter or code if you’re using the Python API**

Especially the converter.inference_input_type and converter.inference_output_type is imporant.
```
k_model = tf.keras.models.load_model(model_path)
converter = tf.lite.TFLiteConverter.from_keras_model(k_model)
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]
converter.inference_input_type = tf.int8
converter.inference_output_type = tf.int8

def representative_data_gen():
    for input_value in data_set:
        yield [input_value.astype(np.float32).reshape((1, 10))]

converter.representative_dataset = representative_data_gen

tf_lite_model_quant = converter.convert()
```

**The output from the converter invocation**

The output of the convertion is the same as without specifying the inference_input and output_type. 
The tflite outputfiles with and without the specification of the inference input type are attached. 
[tflite_conv_test.zip](https://github.com/tensorflow/tensorflow/files/4971633/tflite_conv_test.zip)

**Failure details**
If the conversion is successful, but the generated model is wrong,
state what is wrong:
- I expect a model without the qunatize and the dequantize layer at the beginning and at the end. 
- The generated model has a infernece_input_type of float32 not the expected int8 

**Any other info / logs**

With tensorflow 1.15 the inference_input was int8 of the generated model when specifing the inference input type. Also no quantization or deqauntization layer were putted in the generated model.

**Comment by alanchiao** [comment]
@MeghnaNatraj

**Comment by MeghnaNatraj** [comment]
This feature is available starting TF 2.3. Would you able to upgrade the TF version and try again?

**Comment by FuchsPhi** [comment]
Worked fine with TF 2.3. Thanks.
